crates/anthropic/src/anthropic.rs

```diff
@@ -457,19 +451,16 @@ impl RateLimitInfo {
                 .and_then(|v| v.to_str().ok())
                 .and_then(|v| v.parse::<u64>().ok())
                 .map(Duration::from_secs),
-            requests: RateLimit::from_headers("requests", headers).ok(),
-            tokens: RateLimit::from_headers("tokens", headers).ok(),
-            input_tokens: RateLimit::from_headers("input-tokens", headers).ok(),
-            output_tokens: RateLimit::from_headers("output-tokens", headers).ok(),
+            requests: RateLimit::from_headers("requests", headers),
+            tokens: RateLimit::from_headers("tokens", headers),
+            input_tokens: RateLimit::from_headers("input-tokens", headers),
+            output_tokens: RateLimit::from_headers("output-tokens", headers),
         }
     }
 }
 
-fn get_header<'a>(key: &str, headers: &'a HeaderMap) -> anyhow::Result<&'a str> {
-    Ok(headers
-        .get(key)
-        .with_context(|| format!("missing header `{key}`"))?
```

Comment:
Do we gain anything by removing the `Result` infrastructure from this code?

While we aren't making use of the errors right now, it's much harder to come in and debug code if we're discarding error information early instead of retaining it and discarding it later.